### ISSUE TITLE
geom_bar() broken when used with ylim() or lims(y=) 

### ISSUE THREAD

**Issue by sadanand-singh** [has2k1/plotnine#37]
geom_bar() along with ylim(), or lims(y=()) is broken. If limit does not include 0 in its range, we get a blank plot.  Here is one example:

``` python
df = pd.DataFrame({
    'variable': ['gender', 'gender', 'age', 'age', 'age', 'income', 'income', 'income', 'income'],
    'category': ['Female', 'Male', '1-24', '25-54', '55+', 'Lo', 'Lo-Med', 'Med', 'High'],
    'value': [60, 40, 50, 30, 20, 10, 25, 25, 40],
})
df['variable'] = pd.Categorical(df['variable'], categories=['gender', 'age', 'income'])
dodge_text = position_dodge(width=0.9)                              # new

# Works
(ggplot(df, aes(x='variable', y='value', fill='category'))
 + geom_bar(stat='identity', position='dodge', show_legend=False)   # modified                                            
)

# Works
(ggplot(df, aes(x='variable', y='value', fill='category'))
 + geom_bar(stat='identity', position='dodge', show_legend=False)   # modified            
 + lims(y=(-5, 60)                               
)

# Does NOT work

(ggplot(df, aes(x='variable', y='value', fill='category'))
 + geom_bar(stat='identity', position='dodge', show_legend=False)   # modified            
 + lims(y=(15, 60)                               
)
```

**Comment by has2k1** [comment]
```python
# Does NOT work

(ggplot(df, aes(x='variable', y='value', fill='category'))
 + geom_bar(stat='identity', position='dodge', show_legend=False)   # modified            
 + lims(y=(15, 60)                             
)
```
It works as expected. When you use `lims`, `ylim` or `xlim` you create a scale with the given limits. A scale (real or abstract) cannot represent data outside of its calibration limits. That is what the blank plot represents, the bars cannot be chopped up. *You are not changing the view port*. To play with the view port you change the limits of the coordinate system. e.g.

```python
(ggplot(df, aes(x='variable', y='value', fill='category'))
 + geom_bar(stat='identity', position='dodge', show_legend=False)   # modified 
 + coord_cartesian(ylim=(15, 60))
)
```

**Comment by sadanand-singh** [comment]
Thanks, I did not know that. Well, when I am using coor_cartesian(ylim=(30,60)) I do not see any effect of this on the plot.

Effectively, both of the following give me exactly same plot:

```python
# Case 1
(ggplot(df, aes(x='variable', y='value', fill='category'))
 + geom_bar(stat='identity', position='dodge', show_legend=False)   # modified
)

# Case 2
(ggplot(df, aes(x='variable', y='value', fill='category'))
 + geom_bar(stat='identity', position='dodge', show_legend=False)   # modified 
 + coord_cartesian(ylim=(30, 60))
)
```

**Comment by has2k1** [comment]
That is due to a bug that was fixed in #30.
